aws/src/main/java/org/apache/iceberg/aws/AwsClientProperties.java

```diff
@@ -211,7 +220,6 @@ public <T extends S3CrtAsyncClientBuilder> void applyClientCredentialConfigurati
   public AwsCredentialsProvider credentialsProvider(
       String accessKeyId, String secretAccessKey, String sessionToken) {
     if (refreshCredentialsEnabled && !Strings.isNullOrEmpty(refreshCredentialsEndpoint)) {
-      clientCredentialsProviderProperties.putAll(allProperties);
```

Comment:
This would now be covered by the code above where on `initialize` we propagate `allProperties` to `clientCredentialsProviderProperties`.